backend/src/Service/Subscription/SubscriptionService.php

```diff
@@ -0,0 +1,306 @@
+
+    public function subscriptionIsActive($storeOwner, $subscribeId)
+    {
+        $res = $this->saveFinishAuto($storeOwner, $subscribeId);
+        if($res->carsStatus){
+
+            return SubscriptionConstant::CARS_FINISHED;
+        }
+ 
+        $item = $this->subscriptionManager->subscriptionIsActive($storeOwner, $subscribeId);
+        if ($item) { 
+
+          return  $item['status'];
+        }       
+     }
+
+    // check subscription , if time is finish or order count is finish, change status value to 'finished'
+    public function saveFinishAuto($storeOwner, $subscribeId)
+    {
+        $response = [];
+        //Get full information for the current subscription
+      
+        $remainingOrdersOfPackage = $this->subscriptionManager->getRemainingOrders($storeOwner, $subscribeId);
+        dd($remainingOrdersOfPackage);
+        $countCancelledOrder = $this->subscriptionManager->getCountCancelledOrders($subscribeId);
+        $countActiveCars = $this->subscriptionManager-> getCountActiveCars($storeOwner, $subscribeId);
+        
+        $remainingOrdersOfPackage['remainingOrders']= $remainingOrdersOfPackage['remainingOrderss'] + $countCancelledOrder['countCancelledOrder'];
+      
+        $remainingOrdersOfPackage['countActiveCars'] = $countActiveCars['countActiveCars'];
+        $countDeliveredOrder = $this->subscriptionManager->getCountDeliveredOrders($storeOwner, $subscribeId);
+
+        $remainingOrdersOfPackage['countOrdersDelivered'] = $countDeliveredOrder['countDeliveredOrders'];
+
+        if ($remainingOrdersOfPackage['subscriptionEndDate']) {
+   
+            $startDate =$remainingOrdersOfPackage['subscriptionStartDate'];
+            $endDate = date_timestamp_get($remainingOrdersOfPackage['subscriptionEndDate']);
+            $endDate1 = $remainingOrdersOfPackage['subscriptionEndDate'];
+            
+            $now =date_timestamp_get(new DateTime("now"));
+
+            if ($endDate1->format('y-m-d') != $startDate->format('y-m-d'))  {
+
+            if ( $endDate <= $now)  {
+    
+                $this->updateFinish($remainingOrdersOfPackage['subscriptionID'], 'date finished');
+                if($this->getNextSubscription($ownerID)) {
+                    $this->changeIsFutureToFalse($this->getNextSubscription($ownerID));
+                    }
+                $response[] = ["subscripe finished, date is finished"];
+            }
+
+            if ($remainingOrdersOfPackage['remainingOrders'] == 0)  {
+        
+                $this->updateFinish($remainingOrdersOfPackage['subscriptionID'], 'orders finished');
+                if($this->getNextSubscription($ownerID)) {
+                $this->changeIsFutureToFalse($this->getNextSubscription($ownerID));
+                }
+                $response[] = ["subscripe finished, count Orders is finished"];
+            }
+               
+            if ((int)$remainingOrdersOfPackage['packageCarCount'] - (int)$countActiveCars['countActiveCars'] == 0)  {
```

Comment:
How about to user brackets () for the division calculation then compare the result with 0 after using "===" instead of "==".

Reply:
I haven't used it yet